downloader/src/main/kotlin/Downloader.kt

```diff
@@ -352,3 +316,34 @@ class Downloader {
         return DownloadResult(startTime, outputDirectory, sourceArtifact = target.sourceArtifact)
     }
 }
+
+/**
+ * Consolidate [projects] based on their VcsInfo without taking the path into account. As we store VcsInfo per project
+ * but many project definition files actually reside in different sub-directories of the same VCS working tree, it does
+ * not make sense to download (and scan) all of them individually, not even if doing sparse checkouts. Return a map that
+ * associates packages for projects with distinct VCS working trees to all other projects from the same VCS working
```

Comment:
nit: Should it be 'assoiciate [..] with' instead of 'associate [..] to'?

Reply:
I believe with having Kotlin's `associateWith` function in mind "with" is fine (and correct).

Reply:
So you'll update it? e.g. "trees **to** all other projects from"

Reply:
I was confused because "associates packages for projects with" already says "with", but in another context. Yes, I'll change it.

Reply:
Ah, just noticed now that this text already existed before - so my change request was kind of out of scope.